source/reduce/remove_opname_instruction_reduction_pass.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "remove_opname_instruction_reduction_pass.h"
+#include "remove_instruction_reduction_opportunity.h"
+#include "source/opcode.h"
+#include "source/opt/instruction.h"
+
+namespace spvtools {
+namespace reduce {
+
+using namespace opt;
+
+std::vector<std::unique_ptr<ReductionOpportunity>>
+RemoveOpNameInstructionReductionPass::GetAvailableOpportunities(
+    opt::IRContext* context) const {
+  std::vector<std::unique_ptr<ReductionOpportunity>> result;
+
+  for (auto& inst : context->module()->debugs2()) {
+    if (inst.opcode() == SpvOpName || inst.opcode() == SpvOpMemberName) {
```

Comment:
Just something to watch out for.  Some tools rely on the names of variables to do reflection.  I don't think this will effect you.

Reply:
Thanks for the approval!

Regarding name of variable and reflection: I actually though we might remove an OpName only when it is the only usage of the given id. But then I though that we should not second-guess too much, and keep passes as simple as possible for now. We can still tweak this reduction pass later if needed!

Reply:
I am in favour of not second-guessing and keeping passes simple.

When using the reducer in practice, the user passes in an interestingness function to decide whether a reduced module should be considered interesting.  This can contain custom checks for the user's scenario, such as whether a given name is still declared in a module.